API change: controlplane v1beta2

Dyanngg opened the pull request:
Merging the feature branch into master, which constitutes three PRs that involves using the new controlplane.v1beta2 API:
- Upgrade ExternalEntity to v1alpha2 and controlplane API to v1beta2 (#1405)
- Deprecate GroupMemberPod (#1416)
- Make controlplane NetworkPolicy cluster scoped (#1445)
Some small rebasing is done to resolve merge conflicts with #1301. @GraysonWu has already verified with the feature branch that the rebasing does not affect the functionality of that PR.

Dyanngg commented:
> @Dyanngg please revert the changes to the Github workflows (addition of `api-change-controlplane-v1beta2` to branches being tested) as part of this PR

Thanks for the reminder. Done

Dyanngg commented:
/test-all

antoninbas reviewed (approved):
LGTM - It's a huge diff so I didn't review it in details, but I reviewed the individual PRs previously.

abhiraut commented:
> LGTM - It's a huge diff so I didn't review it in details, but I reviewed the individual PRs previously.



> LGTM - It's a huge diff so I didn't review it in details, but I reviewed the individual PRs previously.

Shall we merge Audit logging(#1216 ) first? @qiyueyao plans to push rebased version this afternoon

Dyanngg commented:
> > LGTM - It's a huge diff so I didn't review it in details, but I reviewed the individual PRs previously.
> 
> > LGTM - It's a huge diff so I didn't review it in details, but I reviewed the individual PRs previously.
> 
> Shall we merge Audit logging(#1216 ) first? @qiyueyao plans to push rebased version this afternoon

Okay I'll wait for that

Dyanngg commented:
Rebased onto #1216 and resolved merge conflicts (mostly v1beta1 and v1beta2 naming conflicts). Added EnableLogging field into `pkg/apis/controlplane/v1beta2/types.go`.

Dyanngg commented:
/test-all
